compilers/concrete-optimizer/concrete-optimizer/src/dag/unparametrized.rs

```diff
@@ -259,6 +260,27 @@ impl<'dag> DagBuilder<'dag> {
         )
     }
 
+    pub fn add_change_partition(
+        &mut self,
+        input: OperatorIndex,
+        src_partition: Option<&ExternalPartition>,
```

Comment:
don't use references here, ask for ownership, the caller will decide itself if it need to be clone or not.